manager-rs/directory-service/src/manager.rs

```diff
@@ -0,0 +1,109 @@
+use sqlx::Row;
+use std::sync::Arc;
+
+use crate::data::{context::DirectoryContext, error::DirectoryError};
+use crate::database;
+
+/*  Manager handles all functionality of the directory service.
+ *  Shares a 1:1 relation with the RPC handler.
+ */
+
+/// Registers a network with the directory.
+/// Stores the networks address and public key.
+pub async fn register(
+    context: Arc<DirectoryContext>,
+    network_id: &str,
+    address: &str,
+    public_key: &Vec<u8>,
+) -> Result<(), DirectoryError> {
+    tracing::info!(
+        "Register called: {:?}-{:?}-{:?}",
+        network_id,
+        address,
+        public_key
+    );
+
+    let mut transaction = context.database_pool.begin().await?;
+    database::networks::upsert(&mut transaction, network_id, address, public_key).await?;
+    transaction.commit().await?;
+
+    Ok(())
+}
+
+/// Looks up a network by id and checks if it has been registered.
+/// Returns the address and public key of the network.
+pub async fn lookup_network(
+    context: Arc<DirectoryContext>,
+    network_id: &str,
+) -> Result<(String, Vec<u8>), DirectoryError> {
+    tracing::info!("Looup network called: {:?}", network_id);
+
+    let mut transaction = context.database_pool.begin().await?;
+    let row = database::networks::get(&mut transaction, network_id).await?;
+    let address = row.try_get::<String, &str>("address")?;
```

Comment:
These calls that require knowledge of the database keys probably should be in the DB module?

Reply:
Good point, I was on the fence about that. I'll leave this one for later, since there is a little discussion around it